cli/src/cmd/test.js

```diff
@@ -189,13 +194,16 @@ export async function testApp({
   iExec worker's ${Math.floor(IEXEC_WORKER_HEAP_SIZE / (1024 * 1024))}Mb memory limit exceeded.
   You must refactor your app to run within the memory limit.`
     );
+    clearTimeout(taskTimeoutWarning);
   } else if (exitCode === 0) {
     spinner.succeed('App docker image ran and exited successfully.');
+    clearTimeout(taskTimeoutWarning);
   } else {
     spinner.warn(
       `App docker image ran but exited with error (Exit code: ${exitCode})
   You may want to check it was intentional`
     );
+    clearTimeout(taskTimeoutWarning);
```

Comment:
I would implement this way to avoid repeating when not necessary and ensure timeout is cleared when the promise rejects
```ts
const { exitCode, outOfMemory } = await runDockerContainer(/*...*/).finally(() => { clearTimout(taskTimeoutWarning) });
```